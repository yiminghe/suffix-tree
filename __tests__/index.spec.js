import SuffixTee from '../src/';

describe('suffix-tree', () => {
  it('works', () => {
    const s = 'abcabxabcd';
    const st = new SuffixTee(s);
    expect(st.printTree()).toMatchSnapshot();
  });
});
