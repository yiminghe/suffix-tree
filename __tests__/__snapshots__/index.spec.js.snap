// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`suffix-tree works 1`] = `
"digraph {
	rankdir = LR;
	edge [arrowsize=0.4,fontsize=10]
	node1 [label=\\"\\",style=filled,fillcolor=lightgrey,shape=circle,width=.1,height=.1];
//------leaves------
	node6 [label=\\"\\",shape=point]
	node11 [label=\\"\\",shape=point]
	node2 [label=\\"\\",shape=point]
	node8 [label=\\"\\",shape=point]
	node13 [label=\\"\\",shape=point]
	node3 [label=\\"\\",shape=point]
	node15 [label=\\"\\",shape=point]
	node4 [label=\\"\\",shape=point]
	node9 [label=\\"\\",shape=point]
	node16 [label=\\"\\",shape=point]
//------internal nodes------
	node5 [label=\\"\\",style=filled,fillcolor=lightgrey,shape=circle,width=.07,height=.07]
	node10 [label=\\"\\",style=filled,fillcolor=lightgrey,shape=circle,width=.07,height=.07]
	node7 [label=\\"\\",style=filled,fillcolor=lightgrey,shape=circle,width=.07,height=.07]
	node12 [label=\\"\\",style=filled,fillcolor=lightgrey,shape=circle,width=.07,height=.07]
	node14 [label=\\"\\",style=filled,fillcolor=lightgrey,shape=circle,width=.07,height=.07]
//------edges------
	node1 -> node5 [label=\\"ab\\",weight=3]
	node5 -> node6 [label=\\"xabcd\\",weight=3]
	node5 -> node10 [label=\\"c\\",weight=3]
	node10 -> node11 [label=\\"d\\",weight=3]
	node10 -> node2 [label=\\"abxabcd\\",weight=3]
	node1 -> node7 [label=\\"b\\",weight=3]
	node7 -> node8 [label=\\"xabcd\\",weight=3]
	node7 -> node12 [label=\\"c\\",weight=3]
	node12 -> node13 [label=\\"d\\",weight=3]
	node12 -> node3 [label=\\"abxabcd\\",weight=3]
	node1 -> node14 [label=\\"c\\",weight=3]
	node14 -> node15 [label=\\"d\\",weight=3]
	node14 -> node4 [label=\\"abxabcd\\",weight=3]
	node1 -> node9 [label=\\"xabcd\\",weight=3]
	node1 -> node16 [label=\\"d\\",weight=3]
//------suffix links------
	node5 -> node7 [label=\\"\\",weight=1,style=dotted]
	node10 -> node12 [label=\\"\\",weight=1,style=dotted]
	node7 -> node1 [label=\\"\\",weight=1,style=dotted]
	node12 -> node14 [label=\\"\\",weight=1,style=dotted]
	node14 -> node1 [label=\\"\\",weight=1,style=dotted]
}"
`;
